src/settings-manager/encrypted-metadata-manager.ts

```diff
@@ -71,7 +71,20 @@ export class EncryptedMetadataManager implements SettingsManager {
       fetchMetadata,
       mutateMetadata,
     });
-    this.encryptionKey = encryptionKey;
+    if (encryptionKey) {
+      this.encryptionKey = encryptionKey;
+    } else {
+      console.warn("Encrypted Metadata Manager secret key has not been set.");
+      if (process.env.NODE_ENV === "production") {
+        console.error("Can't start the application without the secret key.");
```

Comment:
I think the error should be thrown, so it can be caught by tooling etc, process.exit(1) is "an extra"

Reply:
Catching an error with the tooling is an important argument. Changed the implementation.